File: core/ContainerWithMostWater/ContainerWithMostWater.java
package com.dbc.code;

public class ContainerWithMostWater {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1, res = 0;
        while (left < right) {
            res = Math.max(res, (right - left) * Math.min(height[left], height[right]));
            if (height[left] < height[right]) left++;
            else right--;
        }
        return res;
    }
}

File: core/ContainerWithMostWater/ContainerWithMostWater.py
from typing import List


def maxArea(height: List[int]) -> int:
    left, right = 0, len(height) - 1
    res = 0
    while left < right:
        res = max(res, (right - left) * min(height[left], height[right]))
        if height[left] < height[right]:
            left += 1
        else:
            right -= 1

    return res



